tests/stdlib/tosenv.nim

```diff
@@ -0,0 +1,40 @@
+discard """
+  matrix: "--threads"
+  joinable: false
+  targets: "c cpp"
+"""
+import os
+
+block delEnv:
+    const dummyEnvVar = "D20210720T144752" # This env var wouldn't be likely to exist to begin with
+    doAssert (not existsEnv(dummyEnvVar))
+    putEnv(dummyEnvVar, "1")
+    doAssert existsEnv(dummyEnvVar) 
+    delEnv(dummyEnvVar)
+    doAssert (not existsEnv(dummyEnvVar))
+    delEnv(dummyEnvVar) # deleting an already deleted env var
+    doAssert (not existsEnv(dummyEnvVar))
+
+block putEnv:
+    # raises OSError on invalid input
+    doAssertRaises(OSError, putEnv("DUMMY_ENV_VAR_PUT=DUMMY_VALUE",
+            "NEW_DUMMY_VALUE"))
+    doAssertRaises(OSError, putEnv("", "NEW_DUMMY_VALUE"))
+
+block:
+    doAssert getEnv("DUMMY_ENV_VAR_NONEXISTENT", "") == ""
```

Comment:
prefix all env vars with a string derived from module name, eg: `NIM_TESTS_TOSENV_` instead of `"DUMMY_ENV_VAR_` (less chance of conflict)
(or with a uuid-like value as discussed above; either works)